packages/math-input/src/components/keypad-button.tsx

```diff
@@ -35,12 +35,12 @@ interface ReduxProps {
 interface Props extends ReduxProps {
     ariaLabel?: string;
     borders: Border;
-    childKeys: ReadonlyArray<NonManyKeyConfig>;
+    childKeys: ReadonlyArray<KeyConfig>;
```

Comment:
I'm not here right now because it's 6, so I'll do a full review tomorrow. I just want to point that `ReadonlyArray<NonManyKeyConfig>` is intentional here. We don't want a `MANY` key to be a possible childKey, because childKeys are what `MANY` keys render. We don't want a `MANY` key to be able to render a `MANY` key.

Reply:
I knew it was intentional. I didn't remove it because I thought it was a mistake, it just didn't work with the stricter typing, since `Keys.MANY` becomes just the string "MANY" and not a subset of the `Keys` enum the way it seemed to be intended, and the KeyConfig for the ManyKey cant be used as a KeyConfig of any kind. 

Everywhere we used the union type, it blew up because the `string "MANY"` and `enum Keys` are not compatible.

We might be able to get around that by making `Keys` a string literal, but then we can't do exhaustive switches in `ButtonAssets`.

I'll play with it some more tomorrow and see if there's something else we can do here.